src/Helsenorge.Registries/Abstractions/CollaborationProtocolProfile.cs

```diff
@@ -160,7 +160,7 @@ public CollaborationProtocolMessage FindMessageForSender(string messageName)
 
             if (messages == null)
             {
-                return Roles.SelectMany(role => role.SendMessages).FirstOrDefault((m) => m.Name.Equals(messageName, StringComparison.Ordinal));
+                return Roles.SelectMany(role => role.SendMessages).FirstOrDefault((m) => m.Action.Equals(messageName, StringComparison.Ordinal));
```

Comment:
Siden dette er bakoverkompatibelt så skal vel denne fortsette å sammenligne med `m.Name`, ikke `m.Action`?

Reply:
Om `messageName` kan kun inneholde _Message Function_, så vil koden to linjer over fange denne opp og man vil ikke gå inn i if sjekken. Så derfor kan `messageName` også inneholde _AppRec_ eller så ville man ikke hatt den if sjekken. `m.Name` er nå mappet fra `ProcessSpecification.Name` så denne vil ikke inneholde _AppRec_. Nå så mapper vi om _Action_ til `m.Action` og både i nye og gamle kommer _AppRec_ med i _Action_. Forskjellen er at de andre _Action_ i gamle tjenesten vil ha navnet til _Message Function_, mens i nye så vil de ha det faktiske _Action_ navnet.